src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.DebugVerifier.cs

```diff
@@ -119,7 +119,30 @@ private void VerifyExpression(BoundExpression expression, bool overrideSkippedEx
                     return null;
                 }
 
-                return base.VisitCollectionExpression(node);
+                // See NullableWalker.VisitCollectionExpression.getCollectionDetails() which
+                // does not have an element type for the ImplementsIEnumerable case.
+                var hasElementType = node.CollectionTypeKind is not (CollectionExpressionTypeKind.None or CollectionExpressionTypeKind.ImplementsIEnumerable);
+                foreach (var element in node.Elements)
+                {
+                    if (element is BoundCollectionExpressionSpreadElement spread)
+                    {
+                        Visit(spread.Expression);
+                        Visit(spread.Conversion);
+                        if (spread.EnumeratorInfoOpt != null)
+                        {
+                            VisitForEachEnumeratorInfo(spread.EnumeratorInfoOpt);
+                        }
+                        if (hasElementType)
+                        {
+                            Visit(((BoundExpressionStatement?)spread.IteratorBody)?.Expression);
```

Comment:
Do we actually need to dig into the expression here, instead of just visiting the statement, for the verifier to pick it up?

Reply:
`NullableWalker` only visits the expression within the `BoundExpressionStatement` of `spread.IteratorBody`. (The containing `BoundExpressionStatement` is only there to allow sharing code in binding between `foreach` and `..` since the `foreach` infrastructure expects the body to be a statement.)

Reply:
OK, sounds like if we visited the statement here, then we would fail verification because NullableWalker did not also visit that statement.